ginivision/src/main/java/net/gini/android/vision/internal/camera/api/CameraController.java

```diff
@@ -336,24 +337,24 @@ private void configureCamera(Activity activity) {
 
         Camera.Parameters params = mCamera.getParameters();
 
-        List<Camera.Size> previewSizes = params.getSupportedPreviewSizes();
-        Size previewSize = getLargestFourThreeRatioSize(previewSizes);
-        if (previewSize != null) {
-            mPreviewSize = previewSize;
-            params.setPreviewSize(mPreviewSize.width, mPreviewSize.height);
-            LOG.debug("Preview size ({}, {})", mPreviewSize.width, mPreviewSize.height);
-        } else {
-            LOG.warn("No 4:3 preview size found");
-        }
-
         List<Camera.Size> pictureSizes = params.getSupportedPictureSizes();
-        Size pictureSize = getLargestFourThreeRatioSize(pictureSizes);
+        Size pictureSize = getLargestSize(pictureSizes);
         if (pictureSize != null) {
             mPictureSize = pictureSize;
             params.setPictureSize(mPictureSize.width, mPictureSize.height);
             LOG.debug("Picture size ({}, {})", mPictureSize.width, mPictureSize.height);
         } else {
-            LOG.warn("No 4:3 picture size found");
+            LOG.warn("No suitable picture size found");
+        }
+
+        List<Camera.Size> previewSizes = params.getSupportedPreviewSizes();
+        Size previewSize = getLargestSameAspectRatioSize(previewSizes, mPictureSize);
+        if (previewSize != null) {
+            mPreviewSize = previewSize;
+            params.setPreviewSize(mPreviewSize.width, mPreviewSize.height);
+            LOG.debug("Preview size ({}, {})", mPreviewSize.width, mPreviewSize.height);
+        } else {
+            LOG.warn("No suitable preview size found");
```

Comment:
Yes, that is true, but it should never happen (no, really). The CameraApi states [here](https://developer.android.com/reference/android/hardware/Camera.Parameters.html#getSupportedPictureSizes()) and [here](https://developer.android.com/reference/android/hardware/Camera.Parameters.html#getSupportedPreviewSizes()) that there will be least one returned resolution. If there is really no size then there is something seriously wrong with the device.

We also have our requirement checks, which our clients use to verify that the device meets the hardware requirements.